lib/service_manager.dart

```diff
@@ -265,14 +265,19 @@ class ServiceExtensionManager {
 
       if (!_firstFrameEventReceived) {
         bool didSendFirstFrameEvent = false;
-        if (_serviceExtensions.contains(extensions.didSendFirstFrameEvent) ||
-            _pendingServiceExtensions
-                .contains(extensions.didSendFirstFrameEvent)) {
-          final value = await _service.callServiceExtension(
-              extensions.didSendFirstFrameEvent,
-              isolateId: _isolateManager.selectedIsolate.id);
-          didSendFirstFrameEvent =
-              value != null && value.json['enabled'] == 'true';
+        if (isServiceExtensionAvailable(extensions.didSendFirstFrameEvent)) {
+          try {
+            final value = await _service.callServiceExtension(
+                extensions.didSendFirstFrameEvent,
+                isolateId: _isolateManager.selectedIsolate.id);
+            didSendFirstFrameEvent =
+                value != null && value.json['enabled'] == 'true';
+          } catch (e) {
```

Comment:
I don't think the try catch block is needed given the check for isServiceExtensionAvailable.

Reply:
you're right, thanks for the check. removed.